iothub_client/samples/iothub_client_sample_upload_to_blob_custom/iothub_client_sample_upload_to_blob_custom.c

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// CAVEAT: This sample is to demonstrate azure IoT client concepts only and is not a guide design principles or style
+// Checking of return codes and error values shall be omitted for brevity.  Please practice sound engineering practices
+// when writing production code.
+
+#ifndef DONT_USE_UPLOADTOBLOB
+
+#include <stdio.h>
+#include <stdlib.h>
+
+/* This sample uses the _LL APIs of iothub_client for example purposes.
+That does not mean that HTTP only works with the _LL APIs.
+Simply changing the using the convenience layer (functions not having _LL)
+and removing calls to _DoWork will yield the same results. */
+
+#include "iothub.h"
+#include "iothub_device_client.h"
+
+#include "azure_c_shared_utility/shared_util_options.h"
+#include "iothub_message.h"
+#include "iothubtransportmqtt.h"
+
+#ifdef SET_TRUSTED_CERT_IN_SAMPLES
+#include "certs.h"
+#endif // SET_TRUSTED_CERT_IN_SAMPLES
+
+/*String containing Hostname, Device Id & Device Key in the format:                         */
+/*  "HostName=<host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>"                */
+/*  "HostName=<host_name>;DeviceId=<device_id>;SharedAccessSignature=<device_sas_token>"    */
+static const char* connectionString = "[device connection string]";
+
+/*Optional string with http proxy host and integer for http proxy port (Linux only)         */
```

Comment:
really, only linux?

Reply:
I'm not sure, this is what we had in the original multi-block upload to blob sample. Keeping as is for now.